routing/cross_mwm_connector_serialization.hpp

```diff
@@ -387,9 +404,16 @@ class CrossMwmConnectorSerializer final
     buffer.clear();
   }
 
-  static void WriteTransitions(std::vector<Transition> const & transitions,
+  template <typename CrossMwmId>
+  static void WriteTransitions(std::vector<Transition<CrossMwmId>> const & transitions,
                                serial::CodingParams const & codingParams, uint32_t bitsPerOsmId,
-                               uint8_t bitsPerMask, std::vector<uint8_t> & buffer);
+                               uint8_t bitsPerMask, std::vector<uint8_t> & buffer)
+  {
+    MemWriter<vector<uint8_t>> memWriter(buffer);
```

Comment:
decltype(buffer)
?

Reply:
`buffer` has type `std::vector<uint8_t> & buffer`
and
`memWriter` has type `MemWriter<vector<uint8_t>>`.

They are different.

Reply:
```c++
MemWriter<decltype(buffer)> memWriter(buffer);
```
I know that vector and MemWriter are different types :)

Reply:
Oh! Yes, but decltype(buffer) has type `vector<uint8_t> &` but not `vector<uint8_t>`